src/s3/fileViewerManager.ts

```diff
@@ -54,8 +55,9 @@ export class S3FileProvider implements FileProvider {
         const stats = await this.client.headObject({ bucketName: bucket.name, key })
 
         this.updateETag(stats.ETag)
-        this._file.sizeBytes = stats.ContentLength
+        this._file.sizeBytes = stats.ContentLength ?? this._file.sizeBytes
         this._file.lastModified = stats.LastModified
+        this._file.ContentType = stats.ContentType
```

Comment:
now we are using the contenttype given by S3, nice

Reply:
Yeah `downloadObject` returns it too but it's a little harder to get when using the SDKs streams.